File: LeetCode/500-Keyboard Row/solution.cpp
// Author: David
// Email: youchen.du@gmail.com
// Created: 2017-02-11 14:07
// Last modified: 2017-02-11 14:07
// Filename: solution.cpp
// Description:
class Solution {
public:
    vector<string> findWords(vector<string>& words) {
        int rows[3] = {0, 0, 0};
        vector<string> ans;
        vector<string> chars = {"qwertyuiop", "asdfghjkl", "zxcvbnm"};
        for(int i=0; i < chars.size(); i++)
        {
            for(auto c:chars[i])
                rows[i] |= 1<<(c-'a');
        }
        int iword;
        for(auto word:words)
        {
            iword = 0;
            for(auto c:word)
                iword |= 1<<(c-'a');
            for(int i=0; i < 3; i++)
            {
                if((iword|rows[i])==rows[i])
                {
                    ans.push_back(word);
                    break;
                }
            }
        }
        return ans;
    }
};

File: LeetCode/500-Keyboard Row/solution.py
#!/usr/local/bin/python3
# coding: UTF-8
# Author: David
# Email: youchen.du@gmail.com
# Created: 2017-02-11 13:45
# Last modified: 2017-02-11 13:45
# Filename: solution.py
# Description:
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        result = []
        key_groups = ['qwertyuiop', 'asdfghjkl', 'zxcvbnm']
        for word in words:
            for key_group in key_groups:
                if all([c.lower() in key_group for c in word]):
                    result.append(word)
                    break
        return result
